src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1767,136 +1785,25 @@ public function reorderChildren(Node $node)
-                            $value = $this->generateUuid();
-                            break;
-                        case 'jcr:createdBy':
-                        case 'jcr:lastModifiedBy':
-                            $value = (string) $this->credentials->getUserID();
-                            break;
-                        case 'jcr:created':
-                        case 'jcr:lastModified':
-                            $value = new \DateTime();
-                            break;
-                        case 'jcr:etag':
-                            // TODO: http://www.day.com/specs/jcr/2.0/3_Repository_Model.html#3.7.12.1%20mix:etag
-                            $value = 'TODO: generate from binary properties of this node';
-                            break;
-
-                        default:
-                            $defaultValues = $propertyDef->getDefaultValues();
-                            if ($propertyDef->isMultiple()) {
-                                $value = $defaultValues;
-                            } elseif (isset($defaultValues[0])) {
-                                $value = $defaultValues[0];
-                            } else {
-                                // When implementing versionable or activity, we need to handle more properties explicitly
-                                throw new RepositoryException('No default value for autocreated property '.
-                                    $propertyDef->getName(). ' at '.$node->getPath());
-                            }
-                    }
-
-                    $node->setProperty(
-                        $propertyDef->getName(),
-                        $value,
-                        $propertyDef->getRequiredType()
-                    );
-                }
-            } elseif ($propertyDef->isAutoCreated()) {
-                $prop = $node->getProperty($propertyDef->getName());
-                if (!$prop->isModified() && !$prop->isNew()) {
-                    switch($propertyDef->getName()) {
-                        case 'jcr:lastModified':
-                            if ($this->getAutoLastModified()) {
-                                $prop->setValue(new \DateTime());
-                            }
-                            break;
-                        case 'jcr:lastModifiedBy':
-                            if ($this->getAutoLastModified()) {
-                                $prop->setValue($this->credentials->getUserID());
-                            }
-                            break;
-                        case 'jcr:etag':
-                            // TODO: update etag if needed
-                            break;
-                    }
-
-                }
-            }
-        }
-
-        foreach ($def->getDeclaredSupertypes() as $superType) {
-            $this->validateNodeWithType($node, $superType);
-        }
+        $this->nodeProcessor = new NodeProcessor(
+            $this->credentials->getUserID(),
```

Comment:
this was the reason for most of the test failures - we need the id, not the credentials object.